catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswStore.java

```diff
@@ -0,0 +1,344 @@
+                        .text(itemToDelete.toString());
+                filterType = filterAdapter.adapt(filter, cswFilterDelegate);
+                queryConstraintType.setCqlText(CswCqlTextFilter.getInstance()
+                        .getCqlText(filterType));
+                deleteType.setConstraint(queryConstraintType);
+
+                DeleteAction deleteAction = new DeleteAction(deleteType,
+                        DefaultCswRecordMap.getPrefixToUriMapping());
+                transactionRequest.getDeleteActions()
+                        .add(deleteAction);
+            } catch (UnsupportedQueryException e) {
+                throw new IngestException("Unsupported Query.", e);
+            }
+        }
+
+        try {
+            csw.transaction(transactionRequest);
+        } catch (CswException e) {
+            throw new IngestException("Csw Transaction Failed", e);
+        }
+
+        return new DeleteResponseImpl(deleteRequest,
+                properties,
+                new ArrayList(opTrans.getPreviousStateMetacards()));
+    }
+
+    public void setSchemaTransformerManager(TransformerManager schemaTransformerManager) {
+        this.schemaTransformerManager = schemaTransformerManager;
+    }
+
+    private List<Metacard> transactionQuery(List<Filter> idFilters)
+            throws UnsupportedQueryException {
+        Filter createFilter = filterBuilder.allOf(filterBuilder.anyOf(filterBuilder.attribute(
+                Metacard.TAGS)
+                        .is()
+                        .like()
+                        .text(FilterDelegate.WILDCARD_CHAR),
+                filterBuilder.attribute(Metacard.TAGS)
+                        .empty()), filterBuilder.anyOf(idFilters));
+
+        Query query = new QueryImpl(createFilter);
+        QueryRequest queryRequest = new QueryRequestImpl(query);
+
+        SourceResponse sourceResponse = query(queryRequest);
+        List<Result> results = sourceResponse.getResults();
+        return results.stream()
+                .map(Result::getMetacard)
+                .collect(Collectors.toList());
+    }
+
+    private CswTransactionRequest getTransactionRequest() {
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+        return transactionRequest;
+    }
+
+    protected void validateOperation() {
+        Optional result = capabilities.getOperationsMetadata()
+                .getOperation()
+                .stream()
+                .filter(e -> e.getName()
+                        .equals(CswConstants.TRANSACTION))
+                .findFirst();
+        if (!result.isPresent()) {
+            throw new UnsupportedOperationException(
+                    "This CSW Source can't perform the transaction operations required for create/update/delete");
```

Comment:
It would be more appropriate to say that this source doesn't support the Transaction Operation.